microprofile/telemetry/src/main/java/io/helidon/microprofile/telemetry/OpenTelemetryProducer.java

```diff
@@ -67,18 +71,26 @@ class OpenTelemetryProducer {
         this.mpConfig = mpConfig;
     }
 
+    /**
+     * Construct OpenTelemetry. If the OTEL Agent is present, everything is configured and prepared.
+     * We just reuse objects from the Agent.
+     */
     @PostConstruct
     private void init() {
 
         telemetryProperties  = Collections.unmodifiableMap(getTelemetryProperties());
 
         mpConfig.getOptionalValue(EXPORTER_NAME_PROPERTY, String.class).ifPresent(e -> exporterName = e);
 
-        //Initialize OpenTelemetry in a lazy way.
-        if (!isTelemetryDisabled()) {
-            openTelemetry = LazyValue.create(() -> {
+        // If there is an OTEL Agent – delegate everything to it.
+        if (HelidonOpenTelemetry.AgentDetector.isAgentPresent(config)) {
+            openTelemetry =  GlobalOpenTelemetry.get();
+            return;
```

Comment:
Even if there is an agent, we should correctly set Helidon OpenTelemetry tracing